test/parallel/test-fs-open-options.js

```diff
@@ -0,0 +1,258 @@
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+'use strict';
+
+
+// Copied from /test/parallel/test-fs-open.js using options-signature
+(function() {
+  const common = require('../common');
+  const assert = require('assert');
+  const fs = require('fs');
+
+  const tmpdir = require('../common/tmpdir');
+  tmpdir.refresh();
+
+  let caughtException = false;
+
+  try {
+    // Should throw ENOENT, not EBADF
+    // see https://github.com/joyent/node/pull/1228
+    fs.openSync('/8hvftyuncxrt/path/to/file/that/does/not/exist', {
+      flags: 'r'
+    });
+  } catch (e) {
+    assert.strictEqual(e.code, 'ENOENT');
+    caughtException = true;
+  }
+  assert.strictEqual(caughtException, true);
+
+  fs.openSync(__filename);
+
+  fs.open(__filename, common.mustSucceed());
+
+  fs.open(__filename, { flags: 'r' }, common.mustSucceed());
+
+  fs.open(__filename, { flags: 'rs' }, common.mustSucceed());
+
+  fs.open(__filename, { flags: 'r', mode: 0 }, common.mustSucceed());
+
+  fs.open(__filename, { flags: 'r', mode: null }, common.mustSucceed());
+
+  async function promise() {
+    await fs.promises.open(__filename);
+    await fs.promises.open(__filename, { flags: 'r' });
+  }
+
+  promise().then(common.mustCall()).catch(common.mustNotCall());
+
+  assert.throws(
+    () => fs.open(__filename, {
+      flags: 'r',
+      mode: 'boom'
+    }, common.mustNotCall()),
+    {
+      code: 'ERR_INVALID_ARG_VALUE',
+      name: 'TypeError'
+    }
+  );
+
+  for (const extra of [[], ['r'], ['r', 0], ['r', 0, 'bad callback']]) {
+    assert.throws(
+      () => fs.open(__filename, ...extra),
+      {
+        code: 'ERR_INVALID_CALLBACK',
+        name: 'TypeError'
+      }
+    );
+  }
+
+  [false, 1, [], {}, null, undefined].forEach((i) => {
+    assert.throws(
+      () => fs.open(i, { flags: 'r' }, common.mustNotCall()),
+      {
+        code: 'ERR_INVALID_ARG_TYPE',
+        name: 'TypeError'
+      }
+    );
+    assert.throws(
+      () => fs.openSync(i, { flags: 'r' }, common.mustNotCall()),
+      {
+        code: 'ERR_INVALID_ARG_TYPE',
+        name: 'TypeError'
+      }
+    );
+    assert.rejects(
+      fs.promises.open(i, { flags: 'r' }),
+      {
+        code: 'ERR_INVALID_ARG_TYPE',
+        name: 'TypeError'
+      }
+    );
+  });
+
+  // Check invalid modes.
+  [false, [], {}].forEach((mode) => {
+    assert.throws(
+      () => fs.open(__filename, { flags: 'r', mode }, common.mustNotCall()),
+      {
+        code: 'ERR_INVALID_ARG_TYPE'
+      }
+    );
+    assert.throws(
+      () => fs.openSync(__filename, { flags: 'r', mode }, common.mustNotCall()),
+      {
+        code: 'ERR_INVALID_ARG_TYPE'
+      }
+    );
+    assert.rejects(
+      fs.promises.open(__filename, { flags: 'r', mode }),
+      {
+        code: 'ERR_INVALID_ARG_TYPE'
+      }
+    );
+  });
+}());
+
+
+// Copied from /test/parallel/test-fs-open-flags.js using options-signature
+(function() {
+  // Flags: --expose-internals
```

Comment:
```suggestion
```
I don't think this test needs access to internals – if it does, this line should be moved closer to the top of the file.